src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -589,4 +593,90 @@ private static int lastIndexOfNonWhitespace(String line) {
         }
         return length;
     }
+
+    /**
+     * Calculates and returns the type declaration name matching count.
+     *
+     * <p>
+     * Suppose our pattern class is {@code foo.a.b} and class to be matched is
+     * {@code foo.a.ball} then type declaration name matching count would be calculated by
+     * comparing every character, and updating main counter when we hit "." to prevent matching
+     * "a.b" with "a.ball". In this case type declaration name matching count
+     * would be equal to 6 and not 7 (b of ball is not counted).
+     * </p>
+     *
+     * @param patternClass class against which the given class has to be matched
+     * @param classToBeMatched class to be matched
+     * @return class name matching count
+     */
+    public static int typeDeclNameMatchingCount(String patternClass, String classToBeMatched) {
```

Comment:
Why change previous name?

Reply:
The name is changed wrt to FinalClass but not wrt to UnusedLocalVar
https://github.com/checkstyle/checkstyle/blob/2b7942834e9ecba5b21095133c2c4e6295442535/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java#L693-L703

`identifierMatchingCount` is way too broad, this method only works with type declarations name, identifier can be of a normal variable too, the current name looks fine to me, but am open to better names.

Reply:
I would prefer the full name then, since we are moving it to util class for general use if we can't think of something better, i.e. `typeDeclarationNameMatchingCount`, this is still shorter than some other method names in this class. As we move it out of private method, we should consider the name more carefully.

Reply:
`lengthOfMatchInTypeDeclarationName`?

Reply:
The full name will be better, (`typeDeclarationNameMatchingCount`) there are method chains  like `matcher.results().count();`, so `MatchingCount` is good to have in the end. What do you think?

Reply:
Changed to `typeDeclarationNameMatchingCount`

Reply:
> matcher.results().count();

That's another idea; just have method return char array of longest match and call `Stream.of(results).count()` now that we have Java 11 :)

Reply:
I feel it is fine right now, we would have to increase the number of parameters in the function plus the overhead of streams (though it won't bring any drastic performance difference) when we can simply count using the already present `for` loop.

Reply:
Resolved.